### PR TITLE
Update NMS to compatible with both TF & Pytoch models

### PR DESCRIPTION
**Pull request by HectorSVC**

### CODE REVIEW

File: onnxruntime/contrib_ops/cpu/non_max_suppression.cc [hunk 1]
```diff
@@ -37,119 +35,183 @@ void NonMaxSuppression<T>::MaxMin(const T& lhs, const T& rhs, T& min, T& max) co
...
-  }
+  auto& boxes_shape = boxes->Shape();
+  auto& scores_shape = scores->Shape();
+  auto ret = ParepareCompute(boxes_shape, scores_shape);
+  ORT_RETURN_IF_NOT(ret.IsOK(), ret.ErrorMessage());
 
-  const T* boxes_data = boxes->Data<T>();
-  const T* scores_data = scores->Data<T>();
+  const float* boxes_data = boxes->Data<float>();
+  const float* scores_data = scores->Data<float>();
```
Comment (dashesy) [inline onnxruntime/contrib_ops/cpu/non_max_suppression.cc]: Could scores be any type other that float? it comes from Tensor and most of the time is "float" but could be "float16" at one point

Comment (HectorSVC) [reply onnxruntime/contrib_ops/cpu/non_max_suppression.cc]: cpu don't have float 16 kernel. If it is float16, a extra cast operator will be added to cast the data to float, then cast back to float16 for output.

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc [hunk 2]
```diff
@@ -1099,69 +1098,80 @@ with the exception that numpy default keepdims to False instead of True.)DOC")
...
-      .Output(
+orthogonal transformations and translations of the coordinate system; thus translating or reflections of the coordinate system
+result in the same boxes being selected by the algorithm.
+The selected_indices output is a set of integers indexing into the input collection of bounding boxes representing the selected boxes.
+The bounding box coordinates corresponding to the selected indices can then be obtained using the gather operation. This is mostly used in TF models.
+The boxes output is the filtered boxes which set the filtered boxes to [0, 0, 0, 0]. This is mostly used in Pytoch models.)DOC")
+      .Input(
+          0,
+          "boxes",
+          "An input tensor with shape [num_batches, num_classes, spatial_dimension, 4]. The box data format is indicated by center_point_box.",
```
Comment (wenbingl) [inline onnxruntime/core/graph/contrib_ops/contrib_defs.cc]: In most Mask/Faster RCNN, yolo algorithm, the boxes are independent from the class, why is there a num_classes in the box input?

Comment (dashesy) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc]: @wenbingl we regress bounding boxes separately for each class in our Yolo model. I will add the model [here](https://github.com/Microsoft/pytorch_od) soon.

Comment (wenbingl) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc]: @dashesy , cool, you mean the boxes may be different among each class? Do you have any paper or docs to help me refresh my knowledge?

Comment (dashesy) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc]: @wenbingl I looked again! you are correct, that work is not deployed/published yet. @HectorSVC feel free to simplify this, although you already implemented the channel broadcast.

Comment (HectorSVC) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc]: I'll update the schema to remove the class dimension from boxes

**Pull request merged**